src/middleware.js

```diff
@@ -50,7 +60,8 @@ export const multiClientMiddleware = (clients, customMiddlewareOptions) => {
   };
 };
 
-export default (client, customMiddlewareOptions) => {
+export default (client, customMiddlewareOptions, customClientOptions) => {
   const middlewareOptions = { ...defaultOptions, ...customMiddlewareOptions };
-  return multiClientMiddleware({ [middlewareOptions.defaultClientName]: { client } }, middlewareOptions);
+  const options = customClientOptions || {};
+  return multiClientMiddleware({ [middlewareOptions.defaultClientName]: { client, options } }, middlewareOptions);
```

Comment:
Do we need this block? Because middleware and client interceptors are concating at the end, so you don't need to add client options, because If you are using only single client you can define interceptors and whole middleware.
But maybe I'm wrong, I haven't tested it since.